.github/workflows/solidity-foundry.yml

```
@@ -94,3 +119,55 @@ jobs:
           hostname: ${{ secrets.GRAFANA_INTERNAL_HOST }}
           this-job-name: Foundry Tests ${{ matrix.product }}
         continue-on-error: true
+
+  fmt:
+    strategy:
+      fail-fast: false
+      matrix:
+        product: [ ccip ]
+    needs: [ changes ]
+    name: Forge fmt ${{ matrix.product }}
+    # See https://github.com/foundry-rs/foundry/issues/3827
+    runs-on: ubuntu-22.04
+
+    # The if statements for steps after checkout repo is workaround for
+    # passing required check for PRs that don't have filtered changes.
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+        with:
+          submodules: recursive
+
+      # Only needed because we use the NPM versions of packages
+      # and not native Foundry. This is to make sure the dependencies
+      # stay in sync.
+      - name: Setup NodeJS
+        if: needs.changes.outputs.changes == 'true'
+        uses: ./.github/actions/setup-nodejs
+
+      - name: Install Foundry
+        if: needs.changes.outputs.changes == 'true'
+        uses: foundry-rs/foundry-toolchain@8f1998e9878d786675189ef566a2e4bf24869773 # v1.2.0
+        with:
+          version: ${{ env.FOUNDRY_VERSION }}
+
+      - name: Run Forge fmt
+        if: needs.changes.outputs.changes == 'true'
+        run: |
+          forge fmt --check
+        id: fmt
+        working-directory: contracts
+        env:
+          FOUNDRY_PROFILE: ${{ matrix.product }}
+
+      - name: Collect Metrics
+        if: needs.changes.outputs.changes == 'true'
+        id: collect-gha-metrics
+        uses: smartcontractkit/push-gha-metrics-action@dea9b546553cb4ca936607c2267a09c004e4ab3f # v3.0.0
+        with:
+          id: ccip-solidity-foundry-fmt
```

```suggestion
          id: fmt
```

This should match the job name. Since we don't have an explicit one, the job name comes from [line 123](https://github.com/smartcontractkit/chainlink/pull/13941/files#diff-3f9d887c9dbb18630771a38fe1a9d0d0c63c08c79aab722b2942fd4c43b44a56R123). Could also update the job name to match this, if it makes more sense.